src/ui/zcl_abapgit_gui_chunk_lib.clas.abap

```diff
@@ -780,7 +780,11 @@ CLASS zcl_abapgit_gui_chunk_lib IMPLEMENTATION.
           lv_display_url       TYPE zif_abapgit_persistence=>ty_repo-url,
           lv_icon_commit       TYPE string.
 
-    lv_commit_hash = io_repo_online->get_sha1_remote( ).
+    lv_commit_hash = io_repo_online->get_sha1( ).
+    IF lv_commit_hash IS INITIAL.
+      lv_commit_hash = io_repo_online->get_sha1_remote( ).
+    ENDIF.
```

Comment:
what is the difference between sha1 and sha1 remote? the first is a remote commit hash, so is the second?

Reply:
SHA1 can be fetched and set by calling fetch_remote, that's what get_sha1_remote does. get_sha1 just gets the value of SHA1 without fetching it from remote. This is necessary if the user checked out a different commit and set thereby SHA1 to a different value (that of the checked out commit hash).

Reply:
In short get_sha1 gets the local value of mv_branch (which is to be renamed) and get_sha1_remote fetches the current repo hash (commit hash) from remote and sets mv_branch to it.

Reply:
In that case, wouldn't `fetch_sha1_remote` make more sense?

Reply:
can we do the updating/renaming first? I cannot tell what this code does when reading it

switching to a commit or a branch or a tag, I expect to work the same way seen from a user perspective, changing to a branch/tag will refresh the repo

Reply:
sure, will create a related PR - method should be renamed too?

Reply:
SET_SHA1, is this where the commit sha1 is set if the user chooses a commit? Then I suggest it to be renamed to SELECT_COMMIT

SET_BRANCH_NAME is the similar one, but for branches and tags -> SELECT_BRANCH_NAME

GET_BRANCH_NAME -> GET_SELECTED_BRANCH

GET_SHA1 -> GET_SELECTED_COMMIT

And then MV_BRANCH would be MV_CURRENT_COMMIT ?

Reply:
Looks good to me 👍

Reply:
What's with get_sha1_remote?

Reply:
GET_SHA1_REMOTE, it fetches the remote, it could be a branch or tag or commit, so it might not correspond to a branch

GET_SHA1_REMOTE -> GET_CURRENT_REMOTE, would be symmetric to MV_CURRENT_COMMIT, if you like to get rid of the "SHA1" part